### PR TITLE
[SCB-638] Config model mechanism

### PR DESCRIPTION
**Pull request by wujimin**
support get config value by priority or inject priority value to model
```
  @InjectProperties(prefix = "root")
  public static class ConfigWithAnnotation {
    @InjectProperty(prefix = "override", keys = {"high", "low"})
    public String strValue;

    @InjectProperty(keys = "${key}.value")
    public int intValue;

    @InjectProperty(keys = "${list-1}.a.${list-2}.b")
    public long longValue;

    @InjectProperty(keys = "${full-list}")
    public float floatValue;
}
```

### CODE REVIEW

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/impl/PropertyGetter.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    @Override
+    public Long getValue() {
+      return get();
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  static <T> Property<T> getLongProperty(String propName, T defaultValue) {
+    return (Property<T>) new DynamicLongProperty(propName, (Long) defaultValue);
```
Comment (yhs0092) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/impl/PropertyGetter.java]: Maybe we can remove the annotation `@SuppressWarnings("unchecked")` like below?
```java
  static Property<Long> getLongProperty(String propName, Long defaultValue) {
    return new DynamicLongProperty(propName, defaultValue);
  }
```

File: foundations/foundation-config/src/test/java/org/apache/servicecomb/config/inject/TestConfigObjectFactory.java [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+
+  @Test
+  public void placeholder_multi_list() {
+    ConfigWithAnnotation config = new ConfigObjectFactory().create(ConfigWithAnnotation.class,
+        "list-1", Arrays.asList("l1-1", "l1-2"),
+        "list-2", Arrays.asList("l2-1", "l2-2"));
+    // l1-1.a.l2-1.b
+    // l1-2.a.l2-1.b
+    // l1-1.a.l2-2.b
+    // l1-2.a.l2-2.b
```
Comment (yhs0092) [inline foundations/foundation-config/src/test/java/org/apache/servicecomb/config/inject/TestConfigObjectFactory.java]: It seems that the backward placeholders have a higher priority. Does this meets your expectation?

Comment (wujimin) [reply foundations/foundation-config/src/test/java/org/apache/servicecomb/config/inject/TestConfigObjectFactory.java]: it's my expectation, but it's not correct
i'll fix it.

Comment (wujimin) [reply foundations/foundation-config/src/test/java/org/apache/servicecomb/config/inject/TestConfigObjectFactory.java]: done

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/impl/PropertyGetter.java (hunk 1 above)
Comment (wujimin) [reply foundations/foundation-config/src/main/java/org/apache/servicecomb/config/priority/impl/PropertyGetter.java]: done

**Pull request merged**